### PR TITLE
Featur/tags

### PR DESCRIPTION
**Pull request by aleksVoit**
Splitted routes for photos and tags. Added authorization barriers for routes. fixed bugs with delete photos. Added routes for getting all photos of current user and all photos for another user by user.id

### CODE REVIEW

File: src/repository/photos.py [hunk 1]
```diff
@@ -21,61 +36,53 @@ async def create_photo(text: str, url: str, db: Session) -> Photo:
...
 
-async def delete_photo(photo_id: int, db: Session) -> Photo | None:
+async def delete_photo(photo_id: int, user: User, db: Session) -> Type[Photo]:
 
-    photo = db.query(Photo).filter(Photo.id == photo_id).first()
+    photo = db.query(Photo).filter(and_(Photo.id == photo_id, Photo.user_id == user.id)).first()
+    if photo is None:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail='Photo not found')
     if photo:
+        await destroy_cloud_url(photo.url)
```
Comment (oleksyii) [inline src/repository/photos.py:60]: So does it work? It's not critical, could be merged still, but I'm curious.

File: src/repository/tags.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from sqlalchemy.orm import Session
+from src.database.models import Photo, User, Tag
+from src.schemas import PhotoModel, PhotoResponse, TagModel, TagsPhoto
+
+
+from fastapi import UploadFile, HTTPException, status
+
+
+async def add_tags(photo_id: int, body: TagsPhoto, user: User, db: Session) -> Photo | None:
+    print('REPOSITORY add tags')
```
Comment (oleksyii) [inline src/repository/tags.py:13]: Please rm unnecessary prints or use logger

File: src/repository/tags.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    photo = db.query(Photo).filter(and_(Photo.id == photo_id, Photo.user_id == user.id)).first()
+
+    if len(photo.tags) + len(body.tags) > 5:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                            detail=f'Can assign maximum 5 tags for a photo, '
+                                   f'Existing tags: {[tag.name for tag in photo.tags]}')
+    print('REPOSITORY add tags b4 try')
```
Comment (oleksyii) [inline src/repository/tags.py:20]: Rm or use logger

File: src/repository/tags.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    # Appends tags one by one to the photo
+    for tag in body.tags:
+        await add_tag(photo_id, tag, user, db)
+    return photo
+
+
+async def add_tag(photo_id: int, tag: str, user: User, db: Session) -> Type[Photo] | None:
+    print('Repository ADD TAG')
```
Comment (oleksyii) [inline src/repository/tags.py:28]: Rm or use logger

File: src/schemas.py [hunk 5]
```diff
@@ -19,7 +19,7 @@ class UserCreateModel(BaseModel):
 
 class UserLoginModel(BaseModel):
     email: EmailStr
-    hashed_password: str = Field(min_length=6, max_length=255)
+    password: str = Field(min_length=6, max_length=255)
 
 
 class UserDbModel(BaseModel):
```
Comment (oleksyii) [inline src/schemas.py:25]: Might cause merge conflicts. @iamleeon have already changed in his PR

**Review by oleksyii — APPROVED** [review]

**Review by ValKurta — APPROVED** [review]

**Pull request merged**